### ISSUE TITLE
Shutdown Menu is not the right place for Screenshot button 

### ISSUE THREAD

**Issue by cibersheep** [ubports/unity8#331]
Just a reminder of discussed [in its MR](https://github.com/ubports/unity8/pull/329).
- Shutdown menu is not the right place for Screenshot button (button was added as a quick easy solution for PinePhone)
- Instead move it to Session Indicator

**Comment by dobey** [comment]
Whether we think it is "right" place or not, it is consistent with later versions of Android, the platform where vast majority of our customers are coming from.
I would argue that the session indicator is also no place for a screenshot button. It has nothing to do with session management (and this still places it next to the "shut down" options), is significantly more work to get to, and having it be in an indicator would make the code much more complex while also preventing taking a screenshot of any indicator menus.

**Comment by cibersheep** [comment]
Please read «right» as «not in context» with the Power Dialog.
We are not Android. We could argue if their solutions are good for them or not but that might be in an issue report in Android. My husband's Android phone has the take screenshot button in the indicators bar. So our «customers» might come from other Android experience.
Session indicator is where you find Lock, Shutdown, System Settings and About this phone. All related to the system as a whole.

**Comment by UniversalSuperBox** [comment]
Might it make more sense to change the dialog's text instead? No current Android or iOS device calls this the "Power menu", much like Apple no longer calls "the button on the right of the phone" the power button. It's the Side Button now, and it's actually used to trigger Siri instead of power options.

What I mean is that a long press on the Power button doesn't necessarily need to bring only power options up. Maybe the focused dialog isn't the right way to communicate all the options available to the user, and it should no longer be called the "Power menu" or "Shutdown menu"?

**Comment by dobey** [comment]
I have to agree with @UniversalSuperBox here. Instead of sticking to traditional norms and trying to find a better "context" to place the power button in, we would probably be better off re-designing the pop-up so that it can be the right context. Otherwise we're going to end up making a screenshot more difficult to take, with much more work to get to it.

**Comment by cibersheep** [comment]
Well, the thing is: you can call the button and the dialog as you want but is the Dialog that pups up when you pull down from Session indicator > Shutdown
And you find a dialog to Shut down, reboot, screenshot or cancel.

Do you remember a «go to the start button to shutdown»? Or go to «file to set up the format of a pages»?

**Comment by mateosalta** [comment]
I like the idea of an 'app' of sorts in the other post - allowing a temporary key press setup, or other advanced options. power menu was quick and dirty for the pinephone, but we can do better then complicating a purpose built dialog.

**Comment by dobey** [comment]
> Well, the thing is: you can call the button and the dialog as you want but is the Dialog that pups up when you pull down from Session indicator > Shutdown

We can have two separate popups instead though. Just because we are a convergent platform, doesn't mean everything must be exactly the same always on all devices. We can simplify the popup for phones/tablets that happens when long-pressing the side button, which may include power off, reboot, and screenshot, while maintaining a more traditional shutdown dialog for windowed mode access from the indicator or such. We can also support using the standarad screenshot key bindings when using a physical keyboard.

Doing this would handle all these situations, and resolve the concerns about the dialog being a "shutdown dialog."

**Comment by cibersheep** [comment]
That's indeed interesting.
- I was trying to add an entry on the Session indicator to open a Dialog for take a screenshot. That could open to take a screenshot at the moment, or after some seconds, etc
- keyboard prtsc key works already. That's the way I did screenshot with on the PinePhone.
- Is there a button combination that could work on the PinePhone. I know vol+ + vol- doesn't work. Is that true for all two button combinations?
- I would say that power button on a device would be the same as the power button on a computer. I you hold it hard shutdowns the device if you push it

![image](https://user-images.githubusercontent.com/6640041/94963982-36577080-04f9-11eb-9eb1-47dc833d728c.png)

**Comment by dobey** [comment]
> * Is there a button combination that could work on the PinePhone. I know vol+ + vol- doesn't work. Is that true for all two button combinations?

I don't know if there's anything on the pinephone that would work for it. However, whatever button combination is used, should be consistent across all phones, and not limited to phone-specific things here.

> * I would say that power button on a device would be the same as the power button on a computer. I you hold it hard shutdowns the device if you push it

This is clearly not how the power button works on phones though, and people understand this already (heck, even holding the power button in on my PC doesn't do anything, but that's neither here nor there). Also, see the previous comment from Dalton, mentioning how it is no longer even referred to as the "power" button in some newer devices. In reality all the buttons on a phone are soft buttons, and can theoretically be used to do multiple interesting things. For instance, on most phones when you double-press the "power" button, it opens the camera app, and you can press it while in camera as the "shutter" button.

Remember, context is everything, and we shouldn't try to limit the context to defining a soft button as only the power button, but allow using it for multiple interactions that can improve our user experience, and provide features and behavior that people already expect from their phones, to make it easier for people to migrate to UT.

**Comment by cibersheep** [comment]
> [...] However, whatever button combination is used, should be consistent across all phones, and not limited to phone-specific things here.

I agree.

> [...] In reality all the buttons on a phone are soft buttons, and can theoretically be used to do multiple interesting things. 

I see your point. Interesting.
So, we can show the Power Dialog when user shuts down from Session indicator > shutdown and show an Extra features Dialog with «take screen» (for now) when soft button is pressed and hold. 


> Remember, context is everything

That's the beginning of this issue.

> [...] provide features and behavior that people already expect from their phones, to make it easier for people to migrate to UT.

I think we have to improve the user experience form the users we already have first. It would make no sense to add random option to random places just guessing what user from other OS might expect. Do we also add a «set alarm» button to the Power Dialog? A «open camera» button? Calendar?
I think is better to have a consistent OS

**Comment by mateosalta** [comment]
I have an idea - so basically we like the quick access of the button - but are crowding the power off options, so maybe we could place a screenshot button on the screen towards the bottom:

![IMG_20201002_194615](https://user-images.githubusercontent.com/1965279/94979580-26379500-04e9-11eb-8a3f-c0b5678723fe.jpg)

Like a winking yumi within brackets ;)

**Comment by mateosalta** [comment]
Or perhaps we extend the multi purpose to the extreme - make a quick access panel - on off buttons for the most used things, camera shortcut ect.

![1601687640787437629835](https://user-images.githubusercontent.com/1965279/94980103-5fbdcf80-04ec-11eb-8097-6eb9f95f79d3.jpg)
(Design needs work, but basic idea - maybe tangram pieces for buttons instead?)

**Comment by cibersheep** [comment]
> Or perhaps we extend the multi purpose to the extreme - make a quick access panel - on off buttons for the most used things, camera shortcut ect.

I would keep it simple, to be honest. So the soft button menu could be an «additional» menu. A menu of features that cannot be implemented anywhere else.

```
/===============\
| Screenshot    | take a screenshot
|Scrsht  in X s | take a screenshot in X seconds
|Capture session| Start recording the screen
| Check system  | Integrity of the system (SciFi, I don't know if this is even possible. But I hope you get the idea)
\===============/
```

**Comment by dobey** [comment]
> Do we also add a «set alarm» button to the Power Dialog? A «open camera» button? Calendar?

No, but I think we should support standard gestures. All of these things aren't icons that would be included in this popup, but contextual gestures. For instance, in camera app, it should be allowed to use the soft button as a shutter button, for taking a picture. When screen is locked, a quick double tap should open the camera app (and it should be possible to use camera app without unlocking, but only to take new photos/videos and not view the photo roll).

**Comment by dobey** [comment]
> Or perhaps we extend the multi purpose to the extreme - make a quick access panel - on off buttons for the most used things, camera shortcut ect.

This seems like getting into designing a replacement for indicators too, which I think we don't want to do here. :)
